src/scripts/main.js

```diff
@@ -1,3 +1,248 @@
+    },
+    {
+      label: 'Salary',
+      id: 'salary',
+      type: 'number',
+      dataQa: 'salary',
+    },
+  ];
+
+  // Заполнение полей формы
+  fields.forEach((field) => {
+    const label = document.createElement('label');
+
+    label.setAttribute('for', field.id);
+    label.textContent = `${field.label}:`;
+
+    let input;
+
+    if (field.type === 'select') {
+      input = document.createElement('select');
+      input.setAttribute('id', field.id);
+      input.setAttribute('name', field.id);
+      input.setAttribute('required', field.required);
+      input.setAttribute('data-qa', field.dataQa);
+
+      const options = [
+        'Tokyo',
+        'Singapore',
+        'London',
+        'New York',
+        'Edinburgh',
+        'San Francisco',
+      ];
+
+      options.forEach((optionText) => {
+        const option = document.createElement('option');
+
+        option.value = optionText;
+        option.textContent = optionText;
+        input.appendChild(option);
+      });
+    } else {
+      input = document.createElement('input');
+      input.setAttribute('type', field.type);
+      input.setAttribute('id', field.id);
+      input.setAttribute('name', field.id);
+      input.setAttribute('data-qa', field.dataQa);
+    }
+
+    label.appendChild(input);
+    form.appendChild(label);
+  });
+
+  // Кнопка для отправки формы
+  const submitButton = document.createElement('button');
+
+  submitButton.type = 'submit';
+  submitButton.textContent = 'Save to table';
+  form.appendChild(submitButton);
+
+  // Добавляем форму в контейнер
+  formContainer.appendChild(form);
+  tableContainer.appendChild(formContainer);
+
+  // Обработчик события для добавления нового сотрудника
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+
+    // Получаем данные из формы
+    const nam = document.getElementById('name').value.trim();
+    const position = document.getElementById('position').value.trim();
+    const office = document.getElementById('office').value;
+    const age = parseInt(document.getElementById('age').value);
+    const salary = parseInt(document.getElementById('salary').value);
+
+    // Валидация данных
+    let isValid = true;
+
+    clearNotifications(); // Очистка предыдущих уведомлений
+
+    if (nam.length < 4) {
+      showNotification('error', 'Name must have at least 4 characters.');
+      isValid = false;
+    }
+
+    if (!/^[a-zA-Zа-яА-ЯёЁ]+$/.test(nam)) {
+      showNotification('error', 'Name must contain only letters.');
+      isValid = false;
+    }
+
+    if (position === '' || position.length < 4) {
+      showNotification('error', 'Position must have at least 4 characters.');
+      isValid = false;
+    }
+
+    if (isNaN(age)) {
+      showNotification('Age is not a valid number');
+      isValid = false;
+      // здесь можно отобразить уведомление об ошибке
+    } else if (age < 18 || age > 90) {
+      showNotification('Age must be between 18 and 90');
```

Comment:
The 'showNotification' function call is missing the 'error' keyword for the message 'Age must be between 18 and 90'. It should be 'showNotification('error', 'Age must be between 18 and 90');' to ensure the notification is displayed as an error.